ui/phase_manager.py

```
@@ -0,0 +1,344 @@
+        self._phases: list[dict] = []
+        self._phase_lookup: dict[str, dict] = {}
+        self._active_phase: str | None = None
+        self._active_step: str | None = None
+        self._completed_phases: set[str] = set()
+        self._phase_steps: dict[str, list[dict]] = {}
+        self._total_weight: int = 0
+        self._completed_weight: float = 0.0
+        self._session_start: float = 0
+        self._phase_start_times: dict[str, float] = {}
+
+    def start_session(self, phases: list[dict] = None, subject_names: list[str] = None) -> None:
```

Ruff (0.15.9)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)